cmd/action_cmd.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	render "github.com/akuity/kargo-render"
+	libLog "github.com/akuity/kargo-render/internal/log"
+	libOS "github.com/akuity/kargo-render/internal/os"
+	"github.com/akuity/kargo-render/internal/version"
+)
+
+type actionOptions struct{}
+
+func newActionCommand() *cobra.Command {
+	cmdOpts := &actionOptions{}
+
+	return &cobra.Command{
+		Use:    "action",
+		Hidden: true,
+		Args:   cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return cmdOpts.run(cmd.Context(), cmd.OutOrStdout())
+		},
+	}
+}
+
+// run performs manifest rendering in a GitHub Actions-compatible manner.
+func (o *actionOptions) run(_ context.Context, out io.Writer) error {
+	logger := libLog.LoggerOrDie()
```

Comment:
Think this could be moved into `actionOptions`?

Reply:
You mean make `logger` an attribute of `actionOptions`?

I am tentatively ok with that, but where would you propose setting its value?

Reply:
Either set it in the constructor, or pass it from the root command. My main concern with this suggestion was to prepare the command as best as we can for potentially testing, by allowing it to be an injectable dependency.

Reply:
Fixed in 2bdea8b60480c9a951dca3eb7704def5d7f9e465